File: src/main/java/com/bdev/hogwarts_api/controllers_service/EventsControllerService.java
package com.bdev.hogwarts_api.controllers_service;

import com.bdev.hogwarts_api.data.dto.events.Event;
import com.bdev.hogwarts_api.data.dto.events.EventType;

import java.util.Optional;

public interface EventsControllerService {
    Optional<Event> getCurrentEvent(EventType eventType);
}

File: src/main/java/com/bdev/hogwarts_api/controllers_service/EventsControllerService.kt
package com.bdev.hogwarts_api.controllers_service

import com.bdev.hogwarts_api.data.dto.events.Event
import com.bdev.hogwarts_api.data.dto.events.EventType

interface EventsControllerService {
    fun getEvent(id: Long): Event?
    fun getCurrentEvent(eventType: EventType): Event?
    fun getClosesFutureEvents(eventType: EventType, days: Int): List<Event>
}
